src/main/comments.js

```diff
@@ -306,6 +312,61 @@ function attach(comments, ast, text, options) {
   }
 }
 
+const isAllEmptyAndNoLineBreak = (text) => !/[\S\n\u2028\u2029]/.test(text);
+function isOwnLineComment(text, options, decoratedComments, commentIndex) {
+  const { comment, precedingNode } = decoratedComments[commentIndex];
+  const { locStart, locEnd } = options;
+  let start = locStart(comment);
+
+  if (precedingNode) {
```

Comment:
I think this condition isn't needed. It's okay if `precedingNode` is falsy. `currentCommentPrecedingNode !== precedingNode` would still do what we need (we might want to replace `!==` with `!=` though).

Reply:
https://github.com/prettier/prettier/pull/9672#issuecomment-732941426 seems not caused by this part.

```js
a; /*2*/
/*1*/
/*3*/
```

The trailing comments part print like this

```json
[
{"type":"line-suffix","contents":{"type":"concat","parts":[{"type":"concat","parts":[{"type":"line","hard":true},{"type":"break-parent"}]},"","/*4*/"]}},
{"type":"concat","parts":[" ","/*5*/"]},
{"type":"line-suffix","contents":{"type":"concat","parts":[{"type":"concat","parts":[{"type":"line","hard":true},{"type":"break-parent"}]},"","/*6*/"]}}
]
```

Because of `line-suffix`, `/*1*/` print after `/*2*/`

Reply:
It seems it's a separate issue, I'll try to fix it

Reply:
All three comments should be classified as own-line and printed without `line-suffix`. I thought it didn't happen because of the `if (followingNode)` check. Was I wrong?

Reply:
They were attached as trailing even without this PR, because there is no followingNode, problem is in `printTrailingComment`